### PR TITLE
1st solution

### PR DESCRIPTION
**Pull request by obrazi3**

### CODE REVIEW

File: task3/ConvertingIntoNewSystem.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Text;
+using System.Numerics;
+
+namespace Task3
+{
+    ///<summary>
```
Comment (IgorKheidorov) [inline task3/ConvertingIntoNewSystem.cs]: Bad comment, this class is not for work with integer number, convert from one system to another

File: task3/ConvertingIntoNewSystem.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    ///This class contains a method for working with integer numbers
+    ///</summary>
+    class ConvertingIntoNewSystem
+    {
+        ///<summary>
+        ///This method converts a number from a decimal number system to a new number system
+        ///</summary>
+        ///<param name="numberForConverting">Decimal number</param>
```
Comment (IgorKheidorov) [inline task3/ConvertingIntoNewSystem.cs]: What?

File: task3/ConvertingIntoNewSystem.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        ///<param name="radix">New number system</param>
+        ///<returns>Number in the new number system as a string</returns>
+        public string GetConvertedNumber(BigInteger numberForConverting, short radix)
+        {
+            StringBuilder convertedNumberBuffer = new StringBuilder();
+
+            while (numberForConverting > 0)
+            {
```
Comment (IgorKheidorov) [inline task3/ConvertingIntoNewSystem.cs]: Not good idea to make (numberForConverting % radix) operation 3 times, simplify. Can do the same it in 4 lines, DRY

File: task3/ConvertingIntoNewSystem.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Text;
+using System.Numerics;
+
+namespace Task3
+{
+    ///<summary>
+    ///This class contains a method for working with integer numbers
+    ///</summary>
```
Comment (IgorKheidorov) [inline task3/ConvertingIntoNewSystem.cs]: Bad name, this is object, has to has name Convertor

File: task3/EntryPoint.cs [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+            BigInteger numberForConverting;
+            short radix;
+
+            bool checkNumberForConvert = BigInteger.TryParse(args[0], out numberForConverting);
+            bool checkRadix = short.TryParse(args[1], out radix);
+            bool radixRangeCheck = (radix >= 2) && (radix <= 20);
+
+            if (checkNumberForConvert && checkRadix && radixRangeCheck)
+            {
+                ConvertingIntoNewSystem converter = new ConvertingIntoNewSystem();
```
Comment (IgorKheidorov) [inline task3/EntryPoint.cs]: Some message about what do you output is desired

File: task3/Converter.cs [hunk 6]
```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.Text;
+using System.Numerics;
+
+namespace Task3
+{
+    ///<summary>
+    ///This class contains a method for convert from one system to another
```
Comment (AlexanderSoroka) [inline task3/Converter.cs]: Inaccurate class description

File: task3/Converter.cs [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    ///</summary>
+    class Converter
+    {
+        ///<summary>
+        ///This method converts a number from a decimal number system to a new number system
+        ///</summary>
+        ///<returns>Number in the new number system as a string</returns>
+        public string GetConvertedNumber(BigInteger numberForConverting, int radix)
```
Comment (AlexanderSoroka) [inline task3/Converter.cs]: bad method name

File: task3/Converter.cs [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+        {
+            StringBuilder convertedNumberBuffer = new StringBuilder();
+
+            while (numberForConverting > 0)
+            {
+                int remainder = (int)(numberForConverting % radix);
+                if (remainder >= 10)
+                {
+                    char letter = (char)(55 + remainder);
```
Comment (AlexanderSoroka) [inline task3/Converter.cs]: why 55?

File: task3/EntryPoint.cs [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+            try
+            {
+                BigInteger numberForConverting = BigInteger.Parse(args[0]);
+                int radix = int.Parse(args[1]);
+
+                if (args[0][0] == '-')
+                {
+                    throw new Exception("Number must be positive.");
+                }
+                else if (args[0][0] == '0')
```
Comment (AlexanderSoroka) [inline task3/EntryPoint.cs]: what is somebody passes 00001? Surprise!

**Pull request merged**